numba/core/caching.py

```diff
@@ -697,7 +699,15 @@ def _index_key(self, sig, codegen):
         Compute index key for the given signature and codegen.
         It includes a description of the OS and target architecture.
         """
-        return (sig, codegen.magic_tuple())
+        codebytes = self._py_func.__code__.co_code
+        if self._py_func.__closure__ is not None:
+            cvars = tuple([x.cell_contents for x in self._py_func.__closure__])
+        else:
+            cvars = b''
```

Comment:
why changing the type of `cvars`?

Reply:
is that intended for `cvarbytes` instead?

Reply:
fixed.